processor/src/main/java/com/github/nmorel/gwtjackson/rest/processor/RestService.java

```diff
@@ -78,18 +81,23 @@ public RestService( Options options, Element element ) {
 
         Path path = typeElement.getAnnotation( Path.class );
         String baseRestUrl = path.value();
+        
+        Consumes baseConsumes = typeElement.getAnnotation( Consumes.class );
+        Produces baseProduces = typeElement.getAnnotation( Produces.class );
 
         for ( ExecutableElement method : ElementFilter.methodsIn( typeElement.getEnclosedElements() ) ) {
-            parseMethod( baseRestUrl, method );
+            parseMethod( baseRestUrl, baseConsumes, baseProduces, method );
         }
     }
 
-    private void parseMethod( String baseRestUrl, ExecutableElement method ) {
-        AnnotationMirror httpMethodAnnotation = isRestMethod( method );
+    private void parseMethod( String baseRestUrl, Consumes baseConsumes, Produces baseProduces, ExecutableElement method ) {
+        AnnotationMirror httpMethodAnnotation = isRestMethod( method, baseConsumes, baseProduces );
         if ( null == httpMethodAnnotation ) {
             // not a rest method
             return;
         }
+        String consumes = getFirstJsonType( getConsumes( baseConsumes, method ) );
```

Comment:
If so, in lines 169 and 172, it'd change the condition.

For methods with one present `@Consumes` or `@Produces` that does not support json, the headers would be set to `application/json` by default. I think this should not be the case.

Otherwise, I could duplicate the getConsumes/getProduces methods to return either the only json type found, or all types found for the condition.

I admit that this code s**** but I don't really see any other way of doing better (without changing too much code).